language/vm/vm_runtime/vm_runtime_types/src/native_structs/serializer.rs

```diff
@@ -61,3 +39,67 @@ pub(crate) fn deserialize_native(
         )?)),
     }
 }
+
+//TODO remove this. The VM does some crazy stuff when Ser/De Values that is non-standard.
+pub struct SimpleDeserializer {
+    input: Vec<u8>,
+}
```

Comment:
I want to call out this section explicitly. I'm not exactly sure the rational for why the vm needs to hand walk through the deserialization process of a couple of types. Maybe @runtian-zhou can shed some like on this?

I was really hoping to explicitly hide the LCS Serializer and Deserializer structs (not export them from the lcs crate) because if people have the ability to interact with them directly, there is a chance that someone could accidentally do something that violates the LCS spec (like we can see we've already done in a number of places). As such I had to copy a smallish portion of code from the LCS Deserializer here in order to mimic the current behavior.

Reply:
This is indeed strange.

Reply:
writing a quick note here given I am not sure what we are referring to precisely.
In the VM a type is not mapped to a real Rust type but just to a generic common single type.
When we serialize and deserialize we actually write the data out as if it was a true Rust struct.
So we need to go through our common single representation "by hand" to mimic writing that out as if it was a real struct type.
I am sure this will answer nothing but there... I put it here...